In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing 
# Setting work directory and load dataset
import os
os.getcwd()
os. chdir("C:/Users/user/Desktop/Avance/final/")
train = pd.read_csv("Data/Train_Retail.csv")
test = pd.read_csv("Data/Test_Retail.csv")
train.shape, test.shape

((8523, 12), (5681, 11))

In [2]:
# As we have two file train and test. it is better to combine file to avoid repeatation for diffrent file 
train["source"] = 'train'
test["source"] = 'test'
df = pd.concat([train,test],ignore_index = True)

D:\Work_Place\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  after removing the cwd from sys.path.


In [3]:
df.head(3)

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales    Item_Type  \
0          Low Fat           FDA15  249.8092          3735.1380        Dairy   
1          Regular           DRC01   48.2692           443.4228  Soft Drinks   
2          Low Fat           FDN15  141.6180          2097.2700         Meat   

   Item_Visibility  Item_Weight  Outlet_Establishment_Year Outlet_Identifier  \
0         0.016047         9.30                       1999            OUT049   
1         0.019278         5.92                       2009            OUT018   
2         0.016760        17.50                       1999            OUT049   

  Outlet_Location_Type Outlet_Size        Outlet_Type source  
0               Tier 1      Medium  Supermarket Type1  train  
1               Tier 3      Medium  Supermarket Type2  train  
2               Tier 1      Medium  Supermarket Type1  train

In [4]:
# letes have look at the datatypes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14204 entries, 0 to 14203
Data columns (total 13 columns):
Item_Fat_Content             14204 non-null object
Item_Identifier              14204 non-null object
Item_MRP                     14204 non-null float64
Item_Outlet_Sales            8523 non-null float64
Item_Type                    14204 non-null object
Item_Visibility              14204 non-null float64
Item_Weight                  11765 non-null float64
Outlet_Establishment_Year    14204 non-null int64
Outlet_Identifier            14204 non-null object
Outlet_Location_Type         14204 non-null object
Outlet_Size                  10188 non-null object
Outlet_Type                  14204 non-null object
source                       14204 non-null object
dtypes: float64(4), int64(1), object(8)
memory usage: 1.4+ MB


In [5]:
# As we seen we 7 categorical variables (dtype represented by object)
# Lets check unique vaues in each columns 
df.apply(lambda x: len(x.unique()))

Item_Fat_Content                 5
Item_Identifier               1559
Item_MRP                      8052
Item_Outlet_Sales             3494
Item_Type                       16
Item_Visibility              13006
Item_Weight                    416
Outlet_Establishment_Year        9
Outlet_Identifier               10
Outlet_Location_Type             3
Outlet_Size                      4
Outlet_Type                      4
source                           2
dtype: int64

In [6]:
# Explore the categoricals variable to check all categories with frequencies 
# instead of one by one explore all require at one time  
#Filter categorical variables
cat_cols = [x for x in df.dtypes.index if df.dtypes[x]=='object']
cat_cols = [x for x in cat_cols if x not in ['Item_Identifier','Outlet_Identifier','source']]
for col in cat_cols:
    print('\nCategories and Frequency for varible %s'%col)
    print(df[col].value_counts())


Categories and Frequency for varible Item_Fat_Content
Low Fat    8485
Regular    4824
LF          522
reg         195
low fat     178
Name: Item_Fat_Content, dtype: int64

Categories and Frequency for varible Item_Type
Fruits and Vegetables    2013
Snack Foods              1989
Household                1548
Frozen Foods             1426
Dairy                    1136
Baking Goods             1086
Canned                   1084
Health and Hygiene        858
Meat                      736
Soft Drinks               726
Breads                    416
Hard Drinks               362
Others                    280
Starchy Foods             269
Breakfast                 186
Seafood                    89
Name: Item_Type, dtype: int64

Categories and Frequency for varible Outlet_Location_Type
Tier 3    5583
Tier 2    4641
Tier 1    3980
Name: Outlet_Location_Type, dtype: int64

Categories and Frequency for varible Outlet_Size
Medium    4655
Small     3980
High      1553
Name: Outlet_Size, dtype: int6

In [7]:
# Lest have look at the statistics of the data which is very important to validate data 
df.describe()

Item_MRP  Item_Outlet_Sales  Item_Visibility   Item_Weight  \
count  14204.000000        8523.000000     14204.000000  11765.000000   
mean     141.004977        2181.288914         0.065953     12.792854   
std       62.086938        1706.499616         0.051459      4.652502   
min       31.290000          33.290000         0.000000      4.555000   
25%       94.012000         834.247400         0.027036      8.710000   
50%      142.247000        1794.331000         0.054021     12.600000   
75%      185.855600        3101.296400         0.094037     16.750000   
max      266.888400       13086.964800         0.328391     21.350000   

       Outlet_Establishment_Year  
count               14204.000000  
mean                 1997.830681  
std                     8.371664  
min                  1985.000000  
25%                  1987.000000  
50%                  1999.000000  
75%                  2004.000000  
max                  2009.000000

In [8]:
# last one check all the missing values 
df.isnull().sum()

Item_Fat_Content                0
Item_Identifier                 0
Item_MRP                        0
Item_Outlet_Sales            5681
Item_Type                       0
Item_Visibility                 0
Item_Weight                  2439
Outlet_Establishment_Year       0
Outlet_Identifier               0
Outlet_Location_Type            0
Outlet_Size                  4016
Outlet_Type                     0
source                          0
dtype: int64

# What we analyzed from data 

* We have 8523 Row, 13 Columns for trainig and  testing 5681 rows.
* There are 7 caategorical variables in data. one more we added to combine train and test.
* Looking at the caolumncs and cateory of data we had 16 types of items with 1559 unique item sold in accross 10 Outlet
  of 4 different type of market stores lacated in 3 different type of areas.

* In Item_Fat_Content columns has 2 categorical vaiable with three diffrent lables as Low Fat, Regular, LF, reg, low 
  fat. we need to make them as single.

* Lopking at the statistics of data Item_Visibility columns has 0.00 visibility. if the items in outlest is not visible 
  then how it will sale. thats in valid data points we need to validate.

6. In data set two columns has missing values.


Lets do that all one by one 

# Data cleaning and imputation

In [9]:
#  imputing missing value for Item_weight
df.Item_Weight.isnull().sum()

2439

In [10]:
#Fill missing values of weight of Item According to means of Item Identifier
df['Item_Weight'] = df['Item_Weight'].fillna(df.groupby('Item_Identifier')['Item_Weight'].transform('mean'))
df.Item_Weight.isnull().sum()

0

In [11]:
# Filling missing vaalues for Outlet_size
# Hypothesis: Outlet_size depend on Outlet_type and Outlet_type_location

combination = pd.crosstab(df["Outlet_Size"], df["Outlet_Type"])
combination

Outlet_Type  Grocery Store  Supermarket Type1  Supermarket Type2  \
Outlet_Size                                                        
High                     0               1553                  0   
Medium                   0               1550               1546   
Small                  880               3100                  0   

Outlet_Type  Supermarket Type3  
Outlet_Size                     
High                         0  
Medium                    1559  
Small                        0

In [12]:
# from above table we assume that Grocery Store size are small
# so map small size to grossory store 
size = {"Grocery Store":'Small'}
mapping = df.Outlet_Type.map(size)
df.Outlet_Size = df.Outlet_Size.combine_first(mapping)

In [13]:
combination = pd.crosstab(df["Outlet_Size"], df["Outlet_Location_Type"])
combination

Outlet_Location_Type  Tier 1  Tier 2  Tier 3
Outlet_Size                                 
High                       0       0    1553
Medium                  1550       0    3105
Small                   2430    1550     925

In [14]:
# from above table we conclude that tier 2 has small size
# so map small size to tier 2 

size = {"Tier 2":'Small'}
mapping = df.Outlet_Location_Type.map(size)
df.Outlet_Size = df.Outlet_Size.combine_first(mapping)

In [15]:
df.Outlet_Size.isnull().sum()

0

# Data validation 

In [16]:
# Looking at the statistics of data min value  of Item_Visibility show 0
# Lets check how many have items_with_0_visibility
items_with_0_visibility = (df['Item_Visibility'] == 0)
print('items_with_0_visibility: %d'%sum(items_with_0_visibility))

items_with_0_visibility: 879


In [17]:
#fill by corresponding means of Item_Identifiers
import numpy as np
df['Item_Visibility'] = df['Item_Visibility'].replace(0.00000,np.nan)#first fill by nam for simplicity
df['Item_Visibility'] = df['Item_Visibility'].fillna(df.groupby('Item_Identifier')['Item_Visibility'].transform('mean'))

In [18]:
items_with_0_visibility = (df['Item_Visibility'] == 0)
print('items_with_0_visibility: %d'%sum(items_with_0_visibility))

items_with_0_visibility: 0


# Feature Engineering

In [19]:
#Fat_Content showing redudancy of differnt types
df.Item_Fat_Content.value_counts()

Low Fat    8485
Regular    4824
LF          522
reg         195
low fat     178
Name: Item_Fat_Content, dtype: int64

In [20]:
#Now replace LF by Low Fat ,low fat by lf,reg by Regular
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace({'low fat':'Low Fat','reg':'Regular','LF':'Low Fat'})
df.Item_Fat_Content.value_counts()

Low Fat    9185
Regular    5019
Name: Item_Fat_Content, dtype: int64

In [21]:
#no of years outlet is working conditon 
df['Outlet_Years'] = 2019-df['Outlet_Establishment_Year']

In [22]:
df.columns

Index(['Item_Fat_Content', 'Item_Identifier', 'Item_MRP', 'Item_Outlet_Sales',
       'Item_Type', 'Item_Visibility', 'Item_Weight',
       'Outlet_Establishment_Year', 'Outlet_Identifier',
       'Outlet_Location_Type', 'Outlet_Size', 'Outlet_Type', 'source',
       'Outlet_Years'],
      dtype='object')

In [23]:
#Divide into test and train:
train1 = df.loc[df['source']=="train"]
test1 = df.loc[df['source']=="test"]

In [24]:
#Drop unnecessary columns:
test1.drop(['Item_Outlet_Sales','source','Outlet_Establishment_Year'],axis=1,inplace=True)
train1.drop(['source','Outlet_Establishment_Year'],axis=1,inplace=True)

D:\Work_Place\Anaconda3\lib\site-packages\pandas\core\frame.py:3697: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  errors=errors)


In [25]:
train1.head()

Item_Fat_Content Item_Identifier  Item_MRP  Item_Outlet_Sales  \
0          Low Fat           FDA15  249.8092          3735.1380   
1          Regular           DRC01   48.2692           443.4228   
2          Low Fat           FDN15  141.6180          2097.2700   
3          Regular           FDX07  182.0950           732.3800   
4          Low Fat           NCD19   53.8614           994.7052   

               Item_Type  Item_Visibility  Item_Weight Outlet_Identifier  \
0                  Dairy         0.016047         9.30            OUT049   
1            Soft Drinks         0.019278         5.92            OUT018   
2                   Meat         0.016760        17.50            OUT049   
3  Fruits and Vegetables         0.022930        19.20            OUT010   
4              Household         0.014670         8.93            OUT013   

  Outlet_Location_Type Outlet_Size        Outlet_Type  Outlet_Years  
0               Tier 1      Medium  Supermarket Type1            20  
1               Tier 3      Medium  Supermarket Type2            10  
2               Tier 1      Medium  Supermarket Type1            20  
3               Tier 3       Small      Grocery Store            21  
4               Tier 3        High  Supermarket Type1            32

In [26]:
test1.head()

Item_Fat_Content Item_Identifier  Item_MRP    Item_Type  Item_Visibility  \
8523          Low Fat           FDW58  107.8622  Snack Foods         0.007565   
8524          Regular           FDW14   87.3198        Dairy         0.038428   
8525          Low Fat           NCN55  241.7538       Others         0.099575   
8526          Low Fat           FDQ58  155.0340  Snack Foods         0.015388   
8527          Regular           FDY38  234.2300        Dairy         0.118599   

      Item_Weight Outlet_Identifier Outlet_Location_Type Outlet_Size  \
8523       20.750            OUT049               Tier 1      Medium   
8524        8.300            OUT017               Tier 2       Small   
8525       14.600            OUT010               Tier 3       Small   
8526        7.315            OUT017               Tier 2       Small   
8527       13.600            OUT027               Tier 3      Medium   

            Outlet_Type  Outlet_Years  
8523  Supermarket Type1            20  
8524  Supermarket Type1            12  
8525      Grocery Store            21  
8526  Supermarket Type1            12  
8527  Supermarket Type3            34

In [27]:
#Export files as modified versions:
train1.to_csv("Data/train_cleaned.csv",index=False)
test1.to_csv("Data/test_cleaned.csv",index=False)

# Variable Encoding

In [28]:
from sklearn.preprocessing import LabelEncoder

#Item_Identifier and outlet_Identifier are also useful for making prediction
le = LabelEncoder()
df['Outlet'] = le.fit_transform(df['Outlet_Identifier'])
df['Item_Id'] = le.fit_transform(df['Item_Identifier'])
df["Items_Type"] = le.fit_transform(df['Item_Type'])
Encoding_Vars = ['Item_Fat_Content','Outlet_Location_Type','Outlet_Size','Outlet_Type','Outlet','Items_Type']
for i in Encoding_Vars:
    df[i] = le.fit_transform(df[i])
df.dtypes

Item_Fat_Content               int64
Item_Identifier               object
Item_MRP                     float64
Item_Outlet_Sales            float64
Item_Type                     object
Item_Visibility              float64
Item_Weight                  float64
Outlet_Establishment_Year      int64
Outlet_Identifier             object
Outlet_Location_Type           int64
Outlet_Size                    int64
Outlet_Type                    int64
source                        object
Outlet_Years                   int64
Outlet                         int64
Item_Id                        int64
Items_Type                     int64
dtype: object

In [29]:
#One Hot Coding:
df = pd.get_dummies(df, columns=['Item_Fat_Content','Outlet_Location_Type','Outlet_Size','Outlet_Type','Outlet','Items_Type'])
df.dtypes

Item_Identifier               object
Item_MRP                     float64
Item_Outlet_Sales            float64
Item_Type                     object
Item_Visibility              float64
Item_Weight                  float64
Outlet_Establishment_Year      int64
Outlet_Identifier             object
source                        object
Outlet_Years                   int64
Item_Id                        int64
Item_Fat_Content_0             uint8
Item_Fat_Content_1             uint8
Outlet_Location_Type_0         uint8
Outlet_Location_Type_1         uint8
Outlet_Location_Type_2         uint8
Outlet_Size_0                  uint8
Outlet_Size_1                  uint8
Outlet_Size_2                  uint8
Outlet_Type_0                  uint8
Outlet_Type_1                  uint8
Outlet_Type_2                  uint8
Outlet_Type_3                  uint8
Outlet_0                       uint8
Outlet_1                       uint8
Outlet_2                       uint8
Outlet_3                       uint8
O

In [30]:
#Drop the columns which have been converted to different types:
df.drop(['Outlet_Establishment_Year'],axis=1,inplace=True)

In [31]:
#Divide into test and train:
train = df.loc[df['source']=="train"]
test = df.loc[df['source']=="test"]

In [32]:
#Drop unnecessary columns:
test.drop(['Item_Outlet_Sales','source'],axis=1,inplace=True)
train.drop(['source'],axis=1,inplace=True)

D:\Work_Place\Anaconda3\lib\site-packages\pandas\core\frame.py:3697: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  errors=errors)


In [33]:
train.shape, test.shape

((8523, 47), (5681, 46))

In [34]:
#Export files as modified versions:
train.to_csv("Data/train_modified_encoded.csv",index=False)
test.to_csv("Data/test_modified_encoded.csv",index=False)

# Model  Building

In [40]:
# Feature selection
target = 'Item_Outlet_Sales'
IDcol = ['Item_Identifier','Outlet_Identifier','Item_Type']
predictors = [x for x in train.columns if x not in [target]+IDcol]
predictors

['Item_MRP',
 'Item_Visibility',
 'Item_Weight',
 'Outlet_Years',
 'Item_Id',
 'Item_Fat_Content_0',
 'Item_Fat_Content_1',
 'Outlet_Location_Type_0',
 'Outlet_Location_Type_1',
 'Outlet_Location_Type_2',
 'Outlet_Size_0',
 'Outlet_Size_1',
 'Outlet_Size_2',
 'Outlet_Type_0',
 'Outlet_Type_1',
 'Outlet_Type_2',
 'Outlet_Type_3',
 'Outlet_0',
 'Outlet_1',
 'Outlet_2',
 'Outlet_3',
 'Outlet_4',
 'Outlet_5',
 'Outlet_6',
 'Outlet_7',
 'Outlet_8',
 'Outlet_9',
 'Items_Type_0',
 'Items_Type_1',
 'Items_Type_2',
 'Items_Type_3',
 'Items_Type_4',
 'Items_Type_5',
 'Items_Type_6',
 'Items_Type_7',
 'Items_Type_8',
 'Items_Type_9',
 'Items_Type_10',
 'Items_Type_11',
 'Items_Type_12',
 'Items_Type_13',
 'Items_Type_14',
 'Items_Type_15']

In [41]:
# loading dependent and independent variables
x = np.array(train[predictors])
y = np.array(train["Item_Outlet_Sales"])

In [42]:
# Spliting data into traing and testing 
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.20, random_state=None)
X_train.shape, X_test.shape, y_train.shape, y_test.shape

((6818, 43), (1705, 43), (6818,), (1705,))

# Apply LinearRegression

In [61]:
from sklearn.linear_model import LinearRegression

In [62]:
model = LinearRegression()
model.fit(X_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [63]:
model.score(X_train,y_train)

0.5684993911623023

In [66]:
pred = model.predict(X_test)
pred

array([5550.82441723, 1445.13227303, 2378.28383004, ..., 3349.68734725,
       1028.14562308, 1540.45792639])

In [67]:
# Root mean square error
from math import sqrt
root_mean_sq = sqrt(mean_squared_error(y_test,pred))
root_mean_sq

1120.5253948553245

# Apply RandomForestRegressor 

In [43]:
import sklearn.ensemble
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error

In [44]:
# fitting model
RF_Model = RandomForestRegressor()
RF_Model.fit(X_train,y_train)

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_decrease=0.0, min_impurity_split=None,
           min_samples_leaf=1, min_samples_split=2,
           min_weight_fraction_leaf=0.0, n_estimators=10, n_jobs=1,
           oob_score=False, random_state=None, verbose=0, warm_start=False)

In [45]:
# Model accuracy
RF_Model.score(X_train,y_train)

0.919351874846347

In [46]:
pred = RF_Model.predict(X_test)
pred

array([5355.3623 , 1062.35048, 3143.84102, ..., 3040.24254,  838.908  ,
       1231.33052])

In [47]:
# Root mean square error
from math import sqrt
root_mean_sq = sqrt(mean_squared_error(y_test,pred))
root_mean_sq

1172.2184575264466

In [48]:
#  build GridSearchCV 
from sklearn.model_selection import GridSearchCV
from scipy.stats import randint, uniform
from random import randint as sp_randint

In [49]:
# setting parameter for model

param_grid ={"max_depth":[10,None],"n_estimators" : [10,100], "max_features" : [5,6,10,20,30,43], "min_samples_split" : [2,4,10],
            "bootstrap" : [True,False]}
Grid_Src_Model = GridSearchCV(RF_Model, param_grid = param_grid)

In [50]:
# fitting model to the training data

Grid_Src_Model.fit(X_train, y_train)

GridSearchCV(cv=None, error_score='raise',
       estimator=RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='auto', max_leaf_nodes=None,
           min_impurity_decrease=0.0, min_impurity_split=None,
           min_samples_leaf=1, min_samples_split=2,
           min_weight_fraction_leaf=0.0, n_estimators=10, n_jobs=1,
           oob_score=False, random_state=None, verbose=0, warm_start=False),
       fit_params=None, iid=True, n_jobs=1,
       param_grid={'max_depth': [10, None], 'n_estimators': [10, 100], 'max_features': [5, 6, 10, 20, 30, 43], 'min_samples_split': [2, 4, 10], 'bootstrap': [True, False]},
       pre_dispatch='2*n_jobs', refit=True, return_train_score='warn',
       scoring=None, verbose=0)

In [51]:
# model traing score 

Grid_Src_Model.score(X_train, y_train)

0.6906566881576525

In [52]:
# model prediction and mean suqreroot error calculation

predicted = Grid_Src_Model.predict(X_test)
rt_mean_sq = sqrt(mean_squared_error(y_test,predicted))
rt_mean_sq

1086.3011457616158

In [53]:
# lets predict sale for testing data
test.head()

Item_Identifier  Item_MRP    Item_Type  Item_Visibility  Item_Weight  \
8523           FDW58  107.8622  Snack Foods         0.007565       20.750   
8524           FDW14   87.3198        Dairy         0.038428        8.300   
8525           NCN55  241.7538       Others         0.099575       14.600   
8526           FDQ58  155.0340  Snack Foods         0.015388        7.315   
8527           FDY38  234.2300        Dairy         0.118599       13.600   

     Outlet_Identifier  Outlet_Years  Item_Id  Item_Fat_Content_0  \
8523            OUT049            20     1114                   1   
8524            OUT017            12     1078                   0   
8525            OUT010            21     1420                   1   
8526            OUT017            12      817                   1   
8527            OUT027            34     1197                   0   

      Item_Fat_Content_1      ...        Items_Type_6  Items_Type_7  \
8523                   0      ...                   0             0   
8524                   1      ...                   0             0   
8525                   0      ...                   0             0   
8526                   0      ...                   0             0   
8527                   1      ...                   0             0   

      Items_Type_8  Items_Type_9  Items_Type_10  Items_Type_11  Items_Type_12  \
8523             0             0              0              0              0   
8524             0             0              0              0              0   
8525             0             0              0              1              0   
8526             0             0              0              0              0   
8527             0             0              0              0              0   

      Items_Type_13  Items_Type_14  Items_Type_15  
8523              1              0              0  
8524              0              0              0  
8525              0              0              0  
8526              1              0              0  
8527              0              0              0  

[5 rows x 46 columns]

In [56]:
IDcol = ['Item_Identifier','Outlet_Identifier','Item_Type']
test_predictors = [x for x in test.columns if x not in IDcol]

In [57]:
# Lets predict for new testing data 
X = test[test_predictors]
Item_Outlet_Sales = Grid_Src_Model.predict(X)
Item_Outlet_Sales

array([1644.13304469, 1330.11279943,  638.51889915, ..., 1868.27351656,
       3572.22831427, 1310.45669489])

In [58]:
test1.loc[:,'Item_Outlet_Sales'] = Item_Outlet_Sales.tolist()
test1.head()

Item_Fat_Content Item_Identifier  Item_MRP    Item_Type  Item_Visibility  \
8523          Low Fat           FDW58  107.8622  Snack Foods         0.007565   
8524          Regular           FDW14   87.3198        Dairy         0.038428   
8525          Low Fat           NCN55  241.7538       Others         0.099575   
8526          Low Fat           FDQ58  155.0340  Snack Foods         0.015388   
8527          Regular           FDY38  234.2300        Dairy         0.118599   

      Item_Weight Outlet_Identifier Outlet_Location_Type Outlet_Size  \
8523       20.750            OUT049               Tier 1      Medium   
8524        8.300            OUT017               Tier 2       Small   
8525       14.600            OUT010               Tier 3       Small   
8526        7.315            OUT017               Tier 2       Small   
8527       13.600            OUT027               Tier 3      Medium   

            Outlet_Type  Outlet_Years  Item_Outlet_Sales  
8523  Supermarket Type1            20        1644.133045  
8524  Supermarket Type1            12        1330.112799  
8525      Grocery Store            21         638.518899  
8526  Supermarket Type1            12        2493.278612  
8527  Supermarket Type3            34        5936.162178

In [59]:
test1.to_csv("Data/output_file.csv",index=True)